src/Testcontainers.Pulsar/PulsarBuilder.cs

```diff
@@ -0,0 +1,207 @@
+namespace Testcontainers.Pulsar;
+
+/// <inheritdoc cref="ContainerBuilder{TBuilderEntity, TContainerEntity, TConfigurationEntity}" />
+[PublicAPI]
+public sealed class PulsarBuilder : ContainerBuilder<PulsarBuilder, PulsarContainer, PulsarConfiguration>
+{
+    public const string PulsarImage = "apachepulsar/pulsar:3.0.4";
```

Comment:
Support the latest version `3.2` too.

Reply:
Thanks for catching the issue with Pulsar 3.2.2! We'll investigate this further, in the Pulsar community.

In the meantime, using Pulsar version 3.0.4 for TestContainers is a good choice. It's an LTS (Long-Term Support) version, which means it will be supported for a longer period compared to Pulsar 3.2.2 [(feature release with shorter support ending in 3 months).](https://pulsar.apache.org/contribute/release-policy/)

The builder's design for A/B testing makes sense. My local TestContainers tests have been running stably as well.

If you think it's ready. Feel free to merge 👍

Reply:
I would prefer to address the issue right at the beginning of a new module. Fixing it afterwards is usually difficult (without a module owner). While I had a quick look at the issue, it seems like there are two issues:

1) The `/pulsar/secret.key` file is not created when the wait strategy tries to create the auth token. This is simple to fix. Either wait for the file until it exists or catch the exception and handle the error.

2) When I wait until the key exists, and the token gets created successfully, I still receive occasionally: `HTTP ERROR 401 Authentication required`. Once in a while, it works. When the wait strategy does not cache the auth token, it seems less likely to fail. Of course, this is not a proper fix.

Both findings indicate some kind of race-condition, and I am not sure if this will happen with version `3.0.4` as well.

Reply:
Thanks for also looking into it, you provided some very useful suggestions. I'll also look into it now 😄